crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,148 @@
+use validation_common::{Digest, Measurements};
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+pub enum SnpPolicyFlags {
+    Debug,
+    SingleSocket,
+    #[default]
+    SMT,
+}
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+pub enum SnpPlatformInfoFlags {
+    #[default]
+    SME,
+    TSME,
+}
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+pub struct Config {
+    /// Values for `author_key_digest` in the report body, as `Measurements::signer()`
+    /// This is the list of public keys which have signed the signing key of the Enarx binary.
+    /// Values for `measurement` in the report body, as `Measurements::hash()`
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(flatten, default)]
+    pub measurements: Measurements,
+
+    /// Values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub id_key_digest: Vec<Digest>,
+
+    /// Prohibited values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub id_key_digest_blacklist: Vec<Digest>,
+
+    /// Minimum value for `policy.abi_major`.`policy.abi_minor`
+    #[serde(default)]
+    pub abi: VersionReq,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_policy_string")]
+    pub policy_flags: Option<u8>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_platform_string")]
+    pub platform_info_flags: Option<PlatformInfoFlags>,
+}
+
+fn from_policy_string<'de, D>(deserializer: D) -> Result<Option<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Vec<SnpPolicyFlags> = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = FlagSet::<PolicyFlags>::from(PolicyFlags::Reserved);
+
+    for flag in s {
+        match flag {
+            SnpPolicyFlags::Debug => {
+                flags |= PolicyFlags::Debug;
+            }
+            SnpPolicyFlags::SingleSocket => {
+                flags |= PolicyFlags::SingleSocket;
+            }
+            SnpPolicyFlags::SMT => {
+                flags |= PolicyFlags::SMT;
+            }
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
+
+fn from_platform_string<'de, D>(deserializer: D) -> Result<Option<PlatformInfoFlags>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let flags: SnpPlatformInfoFlags = Deserialize::deserialize(deserializer)?;
+
+    Ok(Some(match flags {
+        SnpPlatformInfoFlags::SME => PlatformInfoFlags::SME,
+        SnpPlatformInfoFlags::TSME => PlatformInfoFlags::TSME,
+    }))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::config::Config;
+
+    #[test]
+    fn empty_config() {
+        let result: Result<Config, toml::de::Error> = toml::from_str("");
```

Comment:
Either use suggestion above or just do `let config: Config =...` just like you do below